docs/adr/020-joint-consensus.md

```
@@ -0,0 +1,720 @@
+    Data      []byte
+}
+```
+
+**Backward compatibility:** `Type == 0` (zero value) means `EntryTypeCommand`.
+All existing log entries on disk have no type byte in `Data`; their `Type` field
+remains 0 after WAL replay. No WAL format migration is required.
+
+**WAL serialisation change:** `encodeLogEntry` gains a 1-byte `Type` field after
+the existing header:
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)